code/modules/mob/living/silicon/robot/robot_defense.dm

```diff
@@ -320,9 +320,22 @@ GLOBAL_LIST_INIT(blacklisted_borg_hats, typecacheof(list( //Hats that don't real
 		return
 	switch(severity)
 		if(1)
-			Stun(160)
+			emp_knockout(16 SECONDS)
 		if(2)
-			Stun(60)
+			emp_knockout(6 SECONDS)
+
+/mob/living/silicon/robot/proc/emp_knockout(deciseconds)
+	if(!src)
+		return
+	if(deciseconds)
+		src.set_stat(UNCONSCIOUS)
+		addtimer(CALLBACK(src, PROC_REF(wake_from_emp)), deciseconds,TIMER_UNIQUE | TIMER_OVERRIDE)
+
+/mob/living/silicon/robot/proc/wake_from_emp()
+	if(!src)
+		return
+	src.set_stat(CONSCIOUS)
+	src.update_stat()
```

Comment:
```suggestion
	set_stat(CONSCIOUS)
	update_stat()
```
Once you have added `TIMER_DELETE_ME` you no longer need to check if we stopped existing.